File: dynamic-programming-level-1/day-7/s2.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left_pt = 0
        right_pt = 1
        profit = 0

        while right_pt < len(prices):
            curr = prices[right_pt] - prices[left_pt]
            if prices[right_pt] > prices[left_pt]:
                profit = max(profit, curr)
            else:
                left_pt = right_pt
            right_pt += 1

        return profit

File: dynamic-programming-level-1/day-7/s2.c
int maxProfit(int* prices, int pricesSize){
    int profit = 0, left = 0, right = 1, diff;
    while (right < pricesSize){
        diff = prices[right] - prices[left];
        profit = (diff > profit) ? diff : profit;
        if (diff < 0)
            left = right;
        right++;
    }
    return profit;
}